rfc#0001.md

```diff
@@ -0,0 +1,154 @@
+of the workload, either for reloading ("recycling" a Keep) or to
+transfer to another Keep in the same trust domain.  This caching, if
+under the origincal WPEK, is at the expense of Perfect Forward
+Secrecy.  If a Keep is to transfer the workload to another Keep in
+the same domain, the original Keep MAY encrypt the workload under a new
+WPEK.  If the original WPEK is used, that WPEK will need to be
+established in the new Keep either at provisioning time (e.g. by the
+ECA) or at a later date through key negotiation.  Such functionality
+is considered risky, as the key management required is difficult to
+get right, and provides new options for compromise or
+design/implementation errors.  Any design of such functionality requires
+a new RFC (see "Future possibilities" below).
+
+The Keep MUST NOT accept unencrypted workloads, as there is no way
+to verify their provenance.
+The Keep SHOULD listen on a well-known port (see "Unresolved
+questions").
+If a port has been pre-established, between the Enarx Client Agent and
+the Keep, then the Keep MUST listen on this port,
+and MAY also listen on the well-known port.
+The Keep MAY cache the workload for re-use, or transfer to another
+Keep.  If a Keep allows workloads to be used by another Keep, the
+workload MAY be encrypted under a new WPEK, established between the
+original Keep and new Keep.
+Keeps MAY be implemented to enable the capability to perform
+workload caching.
+Enabling of the capability for a Keep to perform caching MUST be at
+the discretion of the ECA or entities within the ECA's trust domain.
+Enabling of the capability for a Keep to perform caching MUST NOT
+be at the discretion of the host.  The host MUST NOT be able to
+enable caching.
+
+## Reference
+
+This RFC arose from this issue: [Workload (WASM binary) loading
+- connection termination](https://github.com/enarx/enarx/issues/140)
+
+## Drawbacks
+
+No drawbacks currently known. Furthermore, this mechanism is core functionality to Enarx.
+
+## Rationale and alternatives
+
+Alternatives considered:
+
+Enarx client agent encrypts workload, sends to CPU/firmware via Enarx
+host agent. CPU/firmware decrypts workload, loads into Keep. This has
+the advantage that we use an existing network connection, and there's
+no need to establish a connection nor a session key between the Keep
+and the client agent.
+Rejected: this is not supported by all architectures.
+
+Enarx client agent encrypts workload, sends to host agent. Host agent
+sends to Keep via a listening agent in the Keep. Keep decrypts, loads
+and runs. This maintains the host agent's status as a proxy between
+other components, and doesn't make further assumptions about routing
+which is made by 2. 
+Rejected:
+- the host agent should act as an untrusted proxy wherever possible,
+and not include process logic.
+- maintaining multiple versions of host agents is complex from
+a deployment point of view.
```

Comment:
I don't actually understand why process logic is required for the host agent to be a proxy in this way, or why multiple versions of the host agent would be necessary with this design.